api/end_device.proto

```diff
@@ -175,18 +173,79 @@ message EndDeviceVersion {
 }
 
 message MACSettings {
-  bool use_adr = 1 [(gogoproto.customname) = "UseADR"];
+  message DataRateIndexValue {
+    DataRateIndex value = 1;
+  }
+  message PingSlotPeriodValue {
+    PingSlotPeriod value = 1;
+  }
+  message AggregatedDutyCycleValue {
+    AggregatedDutyCycle value = 1;
+  }
+  message RxDelayValue {
+    RxDelay value = 1;
+  }
+
+  // Maximum delay for the device to answer a MAC request or a confirmed downlink frame.
+  // Must be set if the device supports class B.
+  google.protobuf.Duration class_b_timeout = 1 [(gogoproto.stdduration) = true];
+  // Periodicity of the class B ping slot.
+  // Must be set if the device supports class B.
+  PingSlotPeriodValue ping_slot_periodicity = 2;
+  // Data rate index of the class B ping slot.
+  // Must be set if the device supports class B.
+  DataRateIndexValue ping_slot_data_rate_index = 3;
+  // Frequency of the class B ping slot (Hz).
+  // Must be set if the device supports class B.
+  uint64 ping_slot_frequency = 4;
+
+  // Maximum delay for the device to answer a MAC request or a confirmed downlink frame.
+  // Must be set if the device supports class C.
```

Comment:
@htdvisser @johanstokking do we need to enforce this? BE says we should, but it looks like we could also just use a NS default?